pkg/virt-controller/watch/migration.go

```diff
@@ -1687,3 +1751,25 @@ func (c *MigrationController) isMigrationPolicyMatched(vmi *virtv1.VirtualMachin
 	migrationPolicyName := vmi.Status.MigrationState.MigrationPolicyName
 	return migrationPolicyName != nil && *migrationPolicyName != ""
 }
+
+func (c *MigrationController) isMigrationHandedOff(migrationKey string) bool {
+	c.handOffLock.Lock()
+	defer c.handOffLock.Unlock()
+
+	_, isHandedOff := c.handOffMap[migrationKey]
+	return isHandedOff
+}
```

Comment:
This can't just check the map. It also needs to check the ```vmi.Status.MigrationState != nil && vmi.Status.MigrationState.MigrationUID == migration.UID```

The reasoning here is that virt-controller needs to be able to go down and come back up while still being consistent. The `handOffMap` is just to account for the fact that we can't always depend on the informer to be up-to-date. However after a restart, we wait for all informers to sync before reconciling, so the informer woudl be accurate then but the handOffMap would be inaccurate because the previous state was lost in the virt-controller restart.

Reply:
To clarify, do you mean checking if only one of them occurs with an `or` condition? e.g. `isHandedOff || <migration state with relevant UID exists`?

As I see it, the scenario you're speaking of is relevant only to virt-contoller's "warm-up" phase. What do you think about the following idea: when the controller's initialized and caches are synced, we will iterate over them and fill the map up. This would help us getting started with a consistent map. WDYT?

Reply:
the map only exists to close the gap caused by a delayed informer update. The VMI status is the first check. If that doesn't indicate that the hand off occured, then we're using the map as a secondary check that will give us confirmation that there's not a delayed informer.

in the case of a resync due to handler just firing up, the map won't give us any information that the informer won't have because we wait for the informer to sync before starting the reconcile loop.

My thought is that it's not really worth re-populating the map in this scenario since the status will be consistent then.

Reply:
Makes sense, thanks for pointing that out.
Done!